### PR TITLE
Modify leaktest sub to account for COWable folded CONSTs from Perl v5.41.x

### PR DESCRIPTION
**Pull request by richardleach**
Prior to the Perl 5.41 development cycle, running code like `('x' x 1000000)` many times in a loop - and keeping an active reference to the resulting scalars - would cause perl to allocate many separate string buffers, each containing 1000000 "x"s, being active.

In the Perl 5.41 development cycle, constant folded strings like `('x' x 1000000)` now get the `IsCOW` flag set. Consequently, running `('x' x 1000000)` many times now gives multiple scalars all pointing to the _same_ string buffer.

The upshot of this is that if you want to deliberately simulate leakage of a buffer, as per this module's _leak.t_ , you must now perform a string permutation, causing a full copy of the buffer to be made. Not doing so means that some tests fail on recent development releases (and will do so on the next stable release). For example:
* http://www.cpantesters.org/cpan/report/d4bd6670-8e35-11ef-b6e9-5e516e8775ea

The attached patch is one way to do this. `localtime` is appended to the relevant constants, forcing the unique copy to be made. (These changes are backwards compatible with the previous constant behaviour.)

Please let me know if you are able to do a release incorporating these patches or otherwise making the tests compatible with IsCOW folded constants. If you have any queries, questions or concerns, please also let me know.

### CODE REVIEW

File: t/leak.t [hunk 1]
```diff
@@ -260,7 +260,7 @@ sub alert {
 sub leaktest {
     my $test = shift;
     my %arg  = (init=>10, test=>100, max_mem_diff=>288, diag=>0, @_);
-    my $tmp = 'x' x 1000000; undef $tmp;
+    my $tmp = ('x' x 1000000).localhost; undef $tmp;
```
Comment (powerman) [inline t/leak.t:263]: I suppose this is a typo and you meant `localtime`?

**Review by powerman — CHANGES_REQUESTED** [review]

File: t/leak.t (hunk 1 above)
Comment (powerman) [reply t/leak.t:263]: And why not just `time`?
Also, a comment about reason would make this change more clear.

Comment (richardleach) [reply t/leak.t:263]: I'll update this PR later today.

**Review by powerman — APPROVED** [review]

**Comment by powerman** [comment]
Thanks!

**Pull request merged**